tools/clang/unittests/HLSL/PixTest.cpp

```diff
@@ -3267,6 +3276,184 @@ void main()
   VERIFY_IS_TRUE(FoundTheVariable);
 }
 
+class DxcBlobImpl : public IDxcBlob {
+  DXC_MICROCOM_REF_FIELD(m_dwRef)
+  std::string m_data;
+
+public:
+  DxcBlobImpl(std::string d) : m_data(std::move(d)), m_dwRef(1) {}
+
+  DXC_MICROCOM_ADDREF_RELEASE_IMPL(m_dwRef)
+  HRESULT STDMETHODCALLTYPE QueryInterface(REFIID iid, void **ppvObject) {
+    return DoBasicQueryInterface<IDxcBlob>(this, iid, ppvObject);
+  }
+  virtual LPVOID STDMETHODCALLTYPE GetBufferPointer(void) override {
+    return m_data.data();
+  }
+  virtual SIZE_T STDMETHODCALLTYPE GetBufferSize(void) override {
+    return static_cast<SIZE_T>(m_data.size() * sizeof(m_data[0]));
+  }
+};
+
+class DxcIncludeHandlerForInjectedSources : public IDxcIncludeHandler {
```

Comment:
Note: this is the same name (in global scope) used in dxc.cpp in dxclib project.  I don't think there's a collision, but maybe there could be if one were to link to that lib from the test.

Reply:
Indeed. Renamed!